pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -282,23 +350,43 @@ func (e *cloudWatchExecutor) executeLogAlertQuery(ctx context.Context, queryCont
 	return response, nil
 }
 
+type authType int
+
+const (
+	authTypeDefault authType = iota
+	authTypeSharedCreds
+	authTypeKeys
+)
+
 func (e *cloudWatchExecutor) getDSInfo(region string) *datasourceInfo {
 	if region == defaultRegion {
 		region = e.DataSource.JsonData.Get("defaultRegion").MustString()
 	}
 
-	authType := e.DataSource.JsonData.Get("authType").MustString()
-	assumeRoleArn := e.DataSource.JsonData.Get("assumeRoleArn").MustString()
+	atStr := e.DataSource.JsonData.Get("authType").MustString()
+	assumeRoleARN := e.DataSource.JsonData.Get("assumeRoleArn").MustString()
 	externalID := e.DataSource.JsonData.Get("externalId").MustString()
 	decrypted := e.DataSource.DecryptedValues()
 	accessKey := decrypted["accessKey"]
 	secretKey := decrypted["secretKey"]
 
+	at := authTypeDefault
+	switch atStr {
+	case sharedCredsAuthType:
+		at = authTypeSharedCreds
+	case keysAuthType:
+		at = authTypeKeys
+	case defaultAuthType:
+		at = authTypeDefault
+	default:
+		plog.Warn("Unrecognized AWS authentication type %q", atStr)
```

Comment:
This will trigger for any existing data sources having auth type arn configured today. Maybe I'm missing some handling of this somewhere. Is the idea that ARN could be used for any auth type moving forward?

Reply:
@marefr The ARN is supposed to be supplemental to the auth type. I will have to look at how that works in the UI with this PR, but will have to be when I'm back from holidays.

Reply:
In the current version there is an `assume role arn` textbox only visible for that data source. In the original PR I just removed that condition so it is always visible.

But yeah, assuming a role is not an authentication method by itself, but rather what you do with the authentication method. In the current version of Grafana the Assume Role ARN actually uses the default credentials chain (the `default` method in the rework) to assume the role. The idea is to give the end-user the option to choose what credentials are used to assume the given role.

Reply:
Okay so I think what's needed is to treat legacy arn auth type as the default (`authTypeDefault`) if specified auth type isn't  `sharedCredsAuthType`, `keysAuthType` or `defaultAuthType`.

Reply:
I agree that there should be a legacy support for datasources configured with ARN as authentication provider.
In addition to this, the warning message is not correct: `atStr` is not interpolated in the warning message:
`t=2020-09-21T08:49:29+0000 lvl=warn msg="Unrecognized AWS authentication type %q" logger=tsdb.cloudwatch arn=nil LOG15_ERROR="Normalized odd number of arguments by adding nil"
`

Reply:
@papagian Ah yeah, good catch.

Reply:
@papagian Fixed it.

Reply:
Added conversion of auth type "arn" to the default.

Reply:
I've fixed this, falling back to default auth method if "arn" is configured.